modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc

```diff
@@ -0,0 +1,271 @@
++
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+
+  # Other spec fields omitted ...
+  # ...
+
+  config:
+    features:
+      transport-encryption: strict <1>
+----
++
+<1> Configure `transport-encryption` to `strict`.
+
+. Apply the `KnativeEventing` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+== Configure additional CA trust bundles [[configure_addition_ca_trust_bundles]]
+
+By default, Eventing clients trusts the OpenShift CA bundle that you have configured when using a custom PKI for OpenShift, as documented at https://docs.openshift.com/container-platform/4.14/networking/configuring-a-custom-pki.html.
+
+If you need to add additional CA bundles for Eventing, you can do so by creating `ConfigMaps` in the `knative-eventing` namespace with label `networking.knative.dev/trust-bundle: true`:
+
+[IMPORTANT]
+====
+Whenever CA bundles `ConfigMaps` are updated, the Eventing clients will automatically add them to their trusted CA bundles when a new connection is established.
+====
+
+. Create a CA bundle for Eventing:
+
+[source,yaml]
+----
+kind: ConfigMap
+metadata:
+  name: my-org-eventing-bundle <1>
+  namespace: knative-eventing
+  labels:
+    networking.knative.dev/trust-bundle: "true"
+data: <2>
+  ca.crt: ...
+  ca1.crt: ...
+  tls.crt: ...
+----
+
+<1> Use a name that is unlikely to conflict with existing or future Eventing `ConfigMap`.
+<2> All keys containing valid PEM-encoded CA bundles will be trusted by Eventing clients.
+
+== Adding `SelfSigned` `ClusterIssuer` to CA trust bundles
+
+In case you are using a `SelfSigned` `ClusterIssuer` as described in the <<setup_selfsigned_clusterissuer>> section, you can add the CA to the Eventing CA trust bundles by running the following commands:
+
+. Export the CA from the `knative-eventing-ca` secret in the {certmanageroperatorname} namespace, `cert-manager` by default:
++
+[source,terminal]
+----
+$ oc get secret -n cert-manager knative-eventing-ca -o=jsonpath='{.data.ca\.crt}' | base64 -d > ca.crt
+----
+. Create a CA trust bundle in the `knative-eventing` namespace:
++
+[source,terminal]
+----
+$ oc create configmap -n knative-eventing my-org-selfsigned-ca-bundle --from-file=ca.crt
+----
+. Label the `ConfigMap` with `networking.knative.dev/trust-bundle: "true"` label:
++
+[source,terminal]
+----
+$ oc label configmap -n knative-eventing my-org-selfsigned-ca-bundle networking.knative.dev/trust-bundle=true
+----
+
+== Ensure seamless CA rotation
+
+Ensuring seamless CA rotation is essential to avoid service downtime, or to deal with an emergency.
+The following procedure explains how you can seamlessly rotate a CA.
+
+1. Create a CA certificate.
+2. Add the public key to the CA trust bundles as described in the <<configure_addition_ca_trust_bundles>> section.
+3. Ensure that all clients have downloaded the latest set of CA trust bundles.
```

Comment:
```suggestion
3. Ensure that all clients have consumed the latest set of CA trust bundles. Knative Eventing components will automatically reload the changed CA bundles. If you have custom workload consuming trust bundles as well, make sure to reload/restart them accordingly.
```

Reply:
Not sure if this is correct. If not, you should probably add that people need to re-rollout components in `knative-eventing`?